lib/src/configuration.dart

```diff
@@ -383,16 +383,64 @@ class Configuration {
     return pubspec;
   }
 
+  String? _getPubspecVersionWithBuildNumber(Version pubspecVersion) {
+    const prefix = "Warning: version_with_build_number is enabled,";
+    if (pubspecVersion.build.isNotEmpty && pubspecVersion.build[0] is int) {
+      final buildNumber = pubspecVersion.build[0] as int;
+      if (pubspecVersion.major > 65535) {
+        _logger.stderr(
+          '$prefix but the major version ${pubspecVersion.major} is higher than 65535.',
+        );
+        return null;
+      }
+      if (pubspecVersion.minor > 654) {
+        _logger.stderr(
+          '$prefix but the minor version ${pubspecVersion.minor} is higher than 654.',
+        );
+        return null;
+      }
+      if (pubspecVersion.patch > 99) {
+        _logger.stderr(
+          '$prefix but the patch version ${pubspecVersion.patch} is higher than 99.',
```

Comment:
Perhaps include instruction on how to fix for patch version (ie `Please increment the minor version and reset the patch version`) and minor (similar). That should help if someone runs into one of those limits.

Build and major aren't as straightforward

Reply:
Ive added your additional comment:
https://github.com/YehudaKremer/msix/blob/3c17abb1e6c2b02741fc5cc902c4c791850e43e9/lib/src/configuration.dart#L402-L408